superset/views/core.py

```diff
@@ -1634,12 +1634,18 @@ def _set_dash_metadata(dashboard, data):
         dashboard.css = data.get("css")
         dashboard.dashboard_title = data["dashboard_title"]
 
-        if "filter_immune_slices" not in md:
-            md["filter_immune_slices"] = []
         if "timed_refresh_immune_slices" not in md:
             md["timed_refresh_immune_slices"] = []
-        if "filter_immune_slice_fields" not in md:
-            md["filter_immune_slice_fields"] = {}
+
+        if "filter_scopes" in md:
+            md.pop("filter_immune_slices", None)
+            md.pop("filter_immune_slice_fields", None)
+            md["filter_scopes"] = json.loads(data.get("filter_scopes", "{}"))
+        else:
+            if "filter_immune_slices" not in md:
```

Comment:
do we have a migration plan for migrating filters over? when will this happen? will we migrate all the form data even if dashboards don't get visited?

Reply:
yeah, i think migration from backend is good idea. Users will update/add filter scopes for the active dashboards, but old dashboard will never got updated. I can make a migration once this feature got released and stabled. 
For migration, the logic could be very simple: just convert all existed dashboard filters as `global` scope, and add immune charts if they had immune settings in dashboard metadata.